core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt

```diff
@@ -1,18 +1,27 @@
 package com.strumenta.kolasu.model
 
-import com.strumenta.kolasu.mapping.position
-import org.antlr.v4.runtime.ParserRuleContext
+interface WithPosition {
+    val position: Position?
```

Comment:
Would it make sense to be able to change the position, for example to translate the node?
In the RPG parser what I do is parsing a subset of the text and then translating the nodes. Let's suppose I parse only lines 3, I would get nodes which have positions on line 1 (because I only passed one line to the parser). I would then translate them by two lines to get the correct positions

Reply:
It's not mutable in the interface because some classes may compute it automatically. However, subclasses could have an updatePosition method or similar.

Reply:
I think subclasses could make val into var, that should be permitted